packages/nimble-components/src/wafer-map/specs/README.md

```diff
@@ -0,0 +1,213 @@
+  * Mouse hover - this event gets triggered whenever the mouse pointer is hovering any of the die elements within the wafer map. We only must detect this in the nimble component, proper handling will be done in the MicroStrategy wrapper. (Tooltip triggering)
+
+### Anatomy 
+
+Shadow DOM:
+```html
+<template id=WaferMapTemplate>
+	<div class="WaferMapContainer">
+		<svg>
+			<g class="zoomContainer">
+				<g>
+				<circle></circle>
+				<circle></circle>
+				<rect></rect>
+				</g>
+			</g>
+		</svg>
+		<div class=WaferMapArea>
+			<canvas/>
+			<svg>
+				<rect>
+				</rect>
+			</svg>
+		</div>
+	</div>
+</template>
+```
+The first SVG element with the "zoomContainer" class attribute contains two circles and a rectangle.
+* The first circle represents the outline of the wafer map component.
+* The second circle, together with the rectangle is part of a solution which creates the illusion that there is a notch present on the wafer map frame.
+
+The second SVG element with the "WaferMapArea" class attribute represents the area where the dies get rendered.
+The rectangle present here is used as the "highlight" which allows us to identify individual dies in the canvas.
+
+- Slot Names: (none)
+- Host Classes: (none)
+- Slotted Content/Slotted Classes: (none)
+- CSS Parts: (none)
+
+### Angular integration
+
+The `nimble-wafer-map` component does not take user input directly from the webpage where it's embedded. All the inputs must be provided through the constructor of the component so in the current case there is no point using ControlValueAccessors.
+
+### Blazor integration
+
+Blazor support is out of scope for the current implementation.
+
+### Visual Appearance
+
+For the initial component, we are going with the original wafer map appearance. 
+![WaferMap Example](./Resources/wafer_map_example.png)
+
+## Implementation
+
+As mentioned above, this is not a component that we start from scratch, but a visualization that already exists which is outdated and not futureproof. 
+As a first step, we are working on the "bloat removal" which is basically the removal of unwanted but exiting features (legend, tooltip etc.) and dependencies.
+
+Original component: [wafermap](https://dev.azure.com/ni/DevCentral/_git/op-web-portal?path=/libraries/wafermap&version=GBfeature/microServicesSplit)\
+Once the "bloat removal" is finished, the next step is to adapt the current 
+
+### Code submission strategy
+
+In order to make the process of submitting the re-written wafer map component easier for us and for the code reviewers as well, we propose the following bottom-up approach:
+* As an initial step, we submit a skeleton of the `nimble-wafer-map` component into the main branch. This will serve as a starting point on which the complete functionality will be built on.
+* Once the skeleton is in place, we start to work on and submit individual pieces of the functionality so it's not very overwhelming to review the code.
+
+### States
```

Comment:
Another set of states to consider are the common ones used by other components: 
1. `disabled` renders the component dimmed out and turns off interactivity. This seems useful to support.
2. `focus` adds a green box around the component when it has tab focus. This seems useful to support.
3. Most of our components render differently depending on the theme of the page. As you mention, since the client can set specific die colors I don't think we need to change those in response to theme. But the chrome around the wafer map (the circle and notch border) should probably change to be visible on different background colors.
4. Some other common ones are `appearance` for showing different styles like removing the chrome and `read-only` for turning off interactions without making it disabled. I don't think we need either for this component initially.

Reply:
1. For the initial release, the only thing this would disabled is the mouse hover over the dies and with it the tooltip. But we can add this for futureproofing. - added to the spec
2. I agree with this. This could be useful when we implement the Accessibility features in the future. - added to the spec.
3. This make sense, however, we would not like to give the liberty for the user to specify the chrome color manually. It would make more sense if we detect the theme used when rendering the wafer map and switch the color of the chrome to something standard set by us. (e.g. NI green when we use a dark theme and black when we use light color). We will discuss this with the team and include a section on this into the Design section.
4. I agree with this. Maybe we can take advantage of this when the wafer map offers a deeper level of interaction.

Reply:
I agree that autodetecting the theme to set the chrome color would be the right way to go. Note that this might depend on some infrastructure that's needed for #783 too: our color tokens mostly used from CSS right now but you'd need to use them from TS and have a way to be notified when the token value changes because the theme has changed so that you can rerender the border. Should be fairly easy for us to add, just want to mention it to link the work item to this discussion.

Reply:
@jattasNI are you planning to have this implemented in the near future? I saw that the linked task (#787 ) is expected to be implemented in Q4 this year,

Reply:
> It would make more sense if we detect the theme used when rendering the wafer map and switch the color of the chrome to something standard set by us.

> The first SVG element with the "zoomContainer" class attribute contains two circles and a rectangle.
> -   The first circle represents the outline of the wafer map component.
> -   The second circle, together with the rectangle is part of a solution which creates the illusion that there is a notch present on the wafer map frame.

If the chrome is defined by the inline svg then we can style the svg the css using the existing token infrastructure. We do a similar thing to style the icons which are inline svgs with css, example: 

https://github.com/ni/nimble/blob/197370ec27961996d864f6356e8208c55ae4a99a/packages/nimble-components/src/icon-base/styles.ts#L43-L47

Reply:
> @jattasNI are you planning to have this implemented in the near future?

From @rajsite's comment above it seems like we likely won't need it but if it turns out we do then we can increase it's priority. Can't commit to it for sure before we know how big the work will be but we want to make you successful.

Reply:
Sure. If not, we will go with @rajsite's suggestion.